app/models/land/user_agent.rb

```diff
@@ -12,5 +12,11 @@ class UserAgent < ApplicationRecord
     lookup_for :browser,         class_name: Browser
 
     has_many :visits
+
+    belongs_to :device_resolution, optional: true
+    belongs_to :browser_color_preference, optional: true
+    belongs_to :browser, optional: true
+
+    accepts_nested_attributes_for :browser
```

Comment:
This was what was requested in the previous set of comments.

```suggestion
    accepts_nested_attributes_for :browser
    accepts_nested_attributes_for :browser_color_preference
    accepts_nested_attributes_for :device_resolution

```